core/server/services/settings/index.js

```diff
@@ -4,16 +4,108 @@
  */
 
 var SettingsModel = require('../../models/settings').Settings,
-    SettingsCache = require('./cache');
+    SettingsCache = require('./cache'),
+    SettingsLoader = require('./loader'),
+    EnsureSettingsFiles = require('./ensure-settings'),
+    _ = require('lodash'),
+    common = require('../../lib/common'),
+    debug = require('ghost-ignition').debug('services:settings:index');
 
 module.exports = {
     init: function init() {
-        // Update the defaults
-        return SettingsModel.populateDefaults()
-            .then(function (settingsCollection) {
-                // Initialise the cache with the result
-                // This will bind to events for further updates
-                SettingsCache.init(settingsCollection);
+        const knownSettings = this.knownSettings();
+
+        debug('init settings service for:', knownSettings);
+
+        // Make sure that supported settings files are available
+        // inside of the `content/setting` directory
+        return EnsureSettingsFiles(knownSettings)
+            .then(() => {
+                // Update the defaults
+                return SettingsModel.populateDefaults()
+                    .then(function (settingsCollection) {
+                    // Initialise the cache with the result
+                    // This will bind to events for further updates
+                        SettingsCache.init(settingsCollection);
+                    });
+            });
+    },
+
+    /**
+    * Global place to switch on more available settings.
+    */
+    knownSettings: function knownSettings() {
+        return ['routes'];
+    },
+
+    /**
+     * Getter for YAML settings.
+     * Example: `settings.get('routes').then(...)`
+     * will return an Object like this:
+     * {routes: {}, collections: {}, resources: {}}
+     * @param {String} setting type of supported setting.
+     * @returns {Promise<Object>} settingsFile
+     * @description Returns settings object as defined per YAML files in
+     * `/content/settings` directory.
+     */
+    get: function get(setting) {
+        const knownSettings = this.knownSettings();
+
+        // CASE: this should be an edge case and only if internal usage of the
+        // getter is incorrect.
+        if (!setting || _.indexOf(knownSettings, setting) < 0) {
+            return Promise.reject(new common.errors.IncorrectUsageError({
+                message: `Requested setting is not supported: '${setting}'.`,
+                help: `Please use only the supported settings: ${knownSettings}.`
+            }));
+        }
+
+        return SettingsLoader(setting)
+            .then((settingsFile) => {
+                debug('setting loaded and parsed:', settingsFile);
+
+                return settingsFile;
+            }).catch((error) => {
+                // TODO: I think this is unnecessary?
```

Comment:
@kirrg001 I wasn't sure about this here. Can you please double check?

Reply:
if you are not doing anything custom here, you don't have to register catch. We expect that the settings loader returns a specific error.

Reply:
Yes, it does. And I tested it also without the catch and it works the same, but I wasn't 100% sure. So I think it's safe to remove both catches?